crates/uv-toolchain/src/platform.rs

```diff
@@ -1,152 +1,127 @@
-use std::{
-    fmt::{self},
-    str::FromStr,
-};
+use std::fmt::Display;
+use std::ops::Deref;
+use std::{fmt, str::FromStr};
 use thiserror::Error;
 
-/// All supported operating systems.
-#[derive(Debug, Clone, Eq, PartialEq)]
-pub enum Os {
-    Windows,
-    Linux,
-    Macos,
-    FreeBsd,
-    NetBsd,
-    OpenBsd,
-    Dragonfly,
-    Illumos,
-    Haiku,
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("Unknown operating system: {0}")]
+    UnknownOs(String),
+    #[error("Unknown architecture: {0}")]
+    UnknownArch(String),
 }
 
-/// All supported CPU architectures
-#[derive(Debug, Clone, Copy, Eq, PartialEq)]
-pub enum Arch {
-    Aarch64,
-    Armv6L,
-    Armv7L,
-    Powerpc64Le,
-    Powerpc64,
-    X86,
-    X86_64,
-    S390X,
-}
+#[derive(Debug, Eq, PartialEq, Clone, Copy)]
+pub struct Arch(pub(crate) target_lexicon::Architecture);
+
+#[derive(Debug, Eq, PartialEq, Clone, Copy)]
+pub struct Os(pub(crate) target_lexicon::OperatingSystem);
 
 #[derive(Debug, Eq, PartialEq, Clone, Copy)]
 pub enum Libc {
-    Gnu,
-    Musl,
+    Some(target_lexicon::Environment),
     None,
 }
 
-#[derive(Error, Debug)]
-pub enum Error {
-    #[error("Operating system not supported: {0}")]
-    OsNotSupported(String),
-    #[error("Architecture not supported: {0}")]
-    ArchNotSupported(String),
-    #[error("Libc type could not be detected")]
-    LibcNotDetected,
-}
-
-impl fmt::Display for Os {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match *self {
-            Self::Windows => write!(f, "Windows"),
-            Self::Macos => write!(f, "MacOS"),
-            Self::FreeBsd => write!(f, "FreeBSD"),
-            Self::NetBsd => write!(f, "NetBSD"),
-            Self::Linux => write!(f, "Linux"),
-            Self::OpenBsd => write!(f, "OpenBSD"),
-            Self::Dragonfly => write!(f, "DragonFly"),
-            Self::Illumos => write!(f, "Illumos"),
-            Self::Haiku => write!(f, "Haiku"),
+impl Libc {
+    pub(crate) fn from_env() -> Self {
+        match std::env::consts::OS {
+            // TODO(zanieb): On Linux, we use the uv target host to determine the libc variant
```

Comment:
I would go with glibc as default on linux target for the start, otherwise the statically linked builds will behave differently than the glibc builds (and the python-build-standalone builds are currently unusable on musl anyway)

Reply:
Our statically linked builds, you mean?

Reply:
Yes, `uv-x86_64-unknown-linux-gnu.tar.gz` and `uv-x86_64-unknown-linux-musl.tar.gz` would behave differently on the same machine since one has been compile with glibc and thinks that current os and the other has been compile with musl and thinks that's the current os (or none because it's statically compiled, i haven't checked)

Reply:
I guess I feel like that's more reasonable behavior for now than _always_ forcing download of GNU Python builds? Like if we can't determine what your system is using you think we should always fallback to GNU?

Reply:
For actually shipping this as stable, we need libc detection and some kind of handling of the musl python-build-standalone situation.

For the preview phase, most linux users are on glibc (it's more like everyone is on glibc except when deploying in an alpine container) and musl python-build-standalone is broken, so assuming glibc works.

Reply:
> We need libc detection

This can still fail and I think we should have a reasonable fallback behavior. Matching the distribution's target seems more intuitive than falling back to glibc in that case?